### PR TITLE
Add JS Interpreter

### PR DESCRIPTION
**Pull request by ginonott**
This PR will connect blockly and the vm together via redux. ~I know this is pretty ugly looking. I'm thinking a follow up PR could be to add a UI library as per #12 and replace our divs with background-colors with real components.~ I went with Fluent UI. We can always change this later.

Play with it here: https://deploy-preview-23--fruk-simulator.netlify.app/

### CODE REVIEW

**Comment by JamesWP** [comment]
Sounds good, will look later. This will help #20 too

File: src/BlocklyInterface/BlocklyEditor.tsx [hunk 1]
```diff
@@ -45,12 +51,33 @@ export const BlocklyEditor: FunctionComponent<BlocklyEditorProps> = ({
 
   // Initialize blockly and return destruction callback
   useEffect(() => {
+    function handleBlocklyChange(event: BlocklyEvent) {
+      if (!blocklyRef.current) {
+        return;
+      }
+
+      console.log(event);
```
Comment (Casper-Oakley) [inline src/BlocklyInterface/BlocklyEditor.tsx:59]: can probs remove

File: src/JavascriptVM/vm.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+import { AppDispatch } from "../store";
+import { blocklySlice } from "../BlocklyInterface/blocklySlice";
+
+export const getIntepreter = (code: string, dispatch: AppDispatch) => {
+  function initInterpreter(
+    intepreter: Interpreter,
+    globalObject: GlobalObject
+  ) {
+    const highlightBlock = intepreter.createNativeFunction((id: string) => {
+      dispatch(blocklySlice.actions.highlightBlock({ blockId: id }));
```
Comment (Casper-Oakley) [inline src/JavascriptVM/vm.ts:11]: in https://github.com/FRUK-Simulator/blockly-intepretation-example/blob/master/src/interpreterwrapper.ts#L12 , there's an extra bit of state to track when the previous highlightBlock call was fired. This is so the [step button](https://github.com/FRUK-Simulator/blockly-intepretation-example/blob/master/src/interpreterwrapper.ts#L29) will step to the next blockly step, as opposed to the next intepreter step, meaning that hitting step will highlight the next block. It'd be nice to have that here as well

Comment (ginonott) [reply src/JavascriptVM/vm.ts:11]: Great feedback! I was trying to figure out how to correspond the interpreter code with the blockly code! Didn't dawn on me to just wait until the next block was highlighted!

**Comment by JamesWP** [comment]
@ginonott looks a million times better now :), but i still see blocks being highlighted after i press the stop button

**Comment by ginonott** [comment]
> LGTM, though we should think about a solution to the infinite loop for another pr.

Agreed. There's just one more change I want to make before merging this. I want to decouple the JS VM from the controls to it.

**Comment by JamesWP** [comment]
> > LGTM, though we should think about a solution to the infinite loop for another pr.
> 
> Agreed. There's just one more change I want to make before merging this. I want to decouple the JS VM from the controls to it.
> 
> Or I can make that a follow up PR.

i don't mind which, ill let you decide

**Pull request merged**